controllers/numaresourcesoperator_controller_test.go

```diff
@@ -187,6 +187,179 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 		})
 	})
 
+	Context("with node group with MCP selector that matches more than one MCP", func() {
+		It("should update the CR condition to degraded when annotation is not enabled", func() {
+			mcpName1 := "test1"
+			label1 := map[string]string{
+				"test1": "test1",
+				"test":  "common",
+			}
+
+			mcpName2 := "test2"
+			label2 := map[string]string{
+				"test2": "test2",
+				"test":  "common",
+			}
+			ng1 := nropv1.NodeGroup{
+				MachineConfigPoolSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"test": "common"},
+				},
+			}
+
+			nro := testobjs.NewNUMAResourcesOperator(objectnames.DefaultNUMAResourcesOperatorCrName, ng1)
+
+			mcp1 := testobjs.NewMachineConfigPool(mcpName1, label1, &metav1.LabelSelector{MatchLabels: label1}, &metav1.LabelSelector{MatchLabels: label1})
+			mcp2 := testobjs.NewMachineConfigPool(mcpName2, label2, &metav1.LabelSelector{MatchLabels: label2}, &metav1.LabelSelector{MatchLabels: label2})
+
+			var err error
+			reconciler, err := NewFakeNUMAResourcesOperatorReconciler(platform.OpenShift, defaultOCPVersion, nro, mcp1, mcp2)
+			Expect(err).ToNot(HaveOccurred())
+
+			key := client.ObjectKeyFromObject(nro)
+			result, err := reconciler.Reconcile(context.TODO(), reconcile.Request{NamespacedName: key})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{}))
+
+			Expect(reconciler.Client.Get(context.TODO(), key, nro)).ToNot(HaveOccurred())
+			degradedCondition := getConditionByType(nro.Status.Conditions, status.ConditionDegraded)
```

Comment:
please add one or more expectations (and let's add one-line reminder comment) to check the actual reconciliation happened as expected

Reply:
indeed I missed that when I updated the expected behavior to be softer, added in the last push. Thanks for noticing this.